packages/vite-plugin-rsc/src/plugin.ts

```diff
@@ -0,0 +1,208 @@
+import { Plugin, ResolvedConfig, Manifest } from "vite";
+import path from "node:path";
+import { pathToFileURL } from "node:url";
+import { existsSync, readFileSync } from "node:fs";
+interface ASTNode {
+  type: string;
+  start: number;
+  end: number;
+  body?: Array<ASTNode>;
+  id?: ASTNode;
+  expression?: ASTNode;
+  declaration?: ASTNode;
+  declarations?: Array<ASTNode>;
+  name?: string;
+  specifiers?: Array<ASTNode>;
+
+  value?: string;
+  exported?: ASTNode;
+}
+
+/**
+ * Checks if the node has a string literal at the top level that matches the statement
+ */
+const hasPragma = (ast: ASTNode, statement: string) =>
```

Comment:
Checks if the AST node has a top level string literal expression with a certain value.